eth/api.go

```diff
@@ -478,7 +478,9 @@ func (api *PrivateDebugAPI) TraceTransaction(ctx context.Context, txHash common.
 		deadlineCtx, cancel := context.WithTimeout(ctx, timeout)
 		go func() {
 			<-deadlineCtx.Done()
-			tracer.(*ethapi.JavascriptTracer).Stop(&timeoutError{})
+			if deadlineCtx.Err() == context.DeadlineExceeded {
+				tracer.(*ethapi.JavascriptTracer).Stop(&timeoutError{})
+			}
```

Comment:
This is not right. The context can be canceled for reasons other than hitting the deadline and should also be stopped in those cases.